develop/pulumi/index.ts

```diff
@@ -0,0 +1,450 @@
+        "CARTS_DYNAMODB_CREATETABLE=true",
+        "AWS_ACCESS_KEY_ID=key",
+        "AWS_SECRET_ACCESS_KEY=dummy",
+    ],
+    hostname: "carts",
+    image: cartsImageRef,
+    memory: 256,
+    name: "carts",
+    networksAdvanced: [
+        {
+            name: network.name,
+        },
+    ],
+    restart: "always",
+}, { dependsOn: cartsDbContainer });
+
+//Launch a container for the orders service
+const ordersContainer = new docker.Container("orders", {
+    capabilities: {
+        drops: ["ALL"],
+    },
+    envs: [
+        "JAVA_OPTS=-XX:MaxRAMPercentage=75.0 -Djava.security.egd=file:/dev/urandom",
+        "SERVER_TOMCAT_ACCESSLOG_ENABLED=true",
+        "SPRING_PROFILES_ACTIVE=mysql,rabbitmq",
+        "SPRING_DATASOURCE_WRITER_URL=jdbc:mariadb://orders-db:3306/orders",
+        "SPRING_DATASOURCE_WRITER_USERNAME=orders_user",
+        `SPRING_DATASOURCE_WRITER_PASSWORD=${mySqlPassword}`,
+        "SPRING_DATASOURCE_READER_URL=jdbc:mariadb://orders-db:3306/orders",
+        "SPRING_DATASOURCE_READER_USERNAME=orders_user",
+        `SPRING_DATASOURCE_READER_PASSWORD=${mySqlPassword}`,
+        "SPRING_RABBITMQ_HOST=rabbitmq",
+    ],
+    hostname: "orders",
+    image: ordersImageRef,
+    memory: 256,
+    name: "orders",
+    networksAdvanced: [
+        {
+            name: network.name,
+        },
+    ],
+    restart: "always",
+}, { dependsOn: ordersDbContainer });
+
+//Launch a container for the checkout service
+const checkoutContainer = new docker.Container("checkout", {
+    capabilities: {
+        drops: ["ALL"],
+    },
+    envs: [
+        "REDIS_URL=redis://checkout-redis:6379",
+        "ENDPOINTS_ORDERS=http://orders:8080",
+      ],
+    hostname: "checkout",
+    image: checkoutImageRef,
+    memory: 256,
+    name: "checkout",
+    networksAdvanced: [
+        {
+            name: network.name,
+        },
+    ],
+    readOnly: true,
+    restart: "always",
+    tmpfs: {
+        "/tmp": "rw,noexec,nosuid",
+    },
+}, { dependsOn: checkoutRedisContainer });
+
+// Launch a UI container
+const uiContainer = new docker.Container("ui", {
+    capabilities: {
+        drops: ["ALL"],
+    },
+    envs: [
+        "JAVA_OPTS=-XX:MaxRAMPercentage=75.0 -Djava.security.egd=file:/dev/urandom",
+        "SERVER_TOMCAT_ACCESSLOG_ENABLED=true",
+        "ENDPOINTS_CATALOG=http://catalog:8080",
+        "ENDPOINTS_CARTS=http://carts:8080",
+        "ENDPOINTS_ORDERS=http://orders:8080",
+        "ENDPOINTS_CHECKOUT=http://checkout:8080",
+        "ENDPOINTS_ASSETS=http://assets:8080",
+    ],
+    hostname: "ui",
+    image: uiImage.id,
+    memory: 256,
+    name: "ui",
+    networksAdvanced: [
+        {
+            name: network.name,
+        },
+    ],
+    ports: [
+        {
+            internal: 8080,
+            external: 8888,
+        },
+    ],
+    restart: "always",
+}, { dependsOn: [ checkoutContainer, catalogContainer, ordersContainer, assetsContainer, cartsContainer ]});
```

Comment:
When this finished, my first question was _How do I see the running app?_ What do you think about exporting it to make that a little easier? 

```suggestion
}, { dependsOn: [ checkoutContainer, catalogContainer, ordersContainer, assetsContainer, cartsContainer ]});


// Export the frontend URL
export const frontendUrl = uiContainer.ports.apply(ports => `http://localhost:${ports![0].external}`);
```

Reply:
Added this export in fdb6c7c